kernel/aster-nix/src/arch/mod.rs

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#[cfg(target_arch = "x86_64")]
+mod x86;
+
+pub trait LinuxAbi {
+    /// Get number of syscall
+    fn syscall_num(&self) -> usize;
+
+    /// Get return value of syscall
+    fn syscall_ret(&self) -> usize;
+
+    /// Set return value of syscall
+    fn set_syscall_ret(&mut self, ret: usize);
+
+    /// Get syscall args
+    fn syscall_args(&self) -> [usize; 6];
+
+    /// Set function argument
+    fn set_argument(&mut self, index: usize, value: usize);
```

Comment:
Looking at this trait along, it is unclear what arguments this method set. And it looks like this method is only useful for setting the arguments of a signal handler and no other parts of the kernel are interested about it. So I suggest moving this knowledge inside `signal/mod.rs`. Hiding such knowledge from the rest of the system makes the code easier to understand and maintain.

---

Edit: consider adding two modules under `arch`: one is `cpu.rs`, the other is `signal.rs`. And then add appropriate arch-independent abstractions or arch-dependent implementations to them.